File: distribution/python/dqops/client/models/collect_error_samples_result.py
from typing import TYPE_CHECKING, Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..models.dqo_job_status import DqoJobStatus
from ..types import UNSET, Unset

if TYPE_CHECKING:
    from ..models.dqo_queue_job_id import DqoQueueJobId
    from ..models.error_sampler_result import ErrorSamplerResult


T = TypeVar("T", bound="CollectErrorSamplesResult")


@_attrs_define
class CollectErrorSamplesResult:
    """Object returned from the operation that queues a "collect error samples" job. The result contains the job id that
    was started and optionally can also contain the result of collecting the error samples  if the operation was started
    with wait=true parameter to wait for the "collect error samples" job to finish.

        Attributes:
            job_id (Union[Unset, DqoQueueJobId]): Identifies a single job that was pushed to the job queue.
            result (Union[Unset, ErrorSamplerResult]): Returns the result with the summary of the error samples collected.
            status (Union[Unset, DqoJobStatus]):
    """

    job_id: Union[Unset, "DqoQueueJobId"] = UNSET
    result: Union[Unset, "ErrorSamplerResult"] = UNSET
    status: Union[Unset, DqoJobStatus] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        job_id: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.job_id, Unset):
            job_id = self.job_id.to_dict()

        result: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.result, Unset):
            result = self.result.to_dict()

        status: Union[Unset, str] = UNSET
        if not isinstance(self.status, Unset):
            status = self.status.value

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if job_id is not UNSET:
            field_dict["jobId"] = job_id
        if result is not UNSET:
            field_dict["result"] = result
        if status is not UNSET:
            field_dict["status"] = status

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        from ..models.dqo_queue_job_id import DqoQueueJobId
        from ..models.error_sampler_result import ErrorSamplerResult

        d = src_dict.copy()
        _job_id = d.pop("jobId", UNSET)
        job_id: Union[Unset, DqoQueueJobId]
        if isinstance(_job_id, Unset):
            job_id = UNSET
        else:
            job_id = DqoQueueJobId.from_dict(_job_id)

        _result = d.pop("result", UNSET)
        result: Union[Unset, ErrorSamplerResult]
        if isinstance(_result, Unset):
            result = UNSET
        else:
            result = ErrorSamplerResult.from_dict(_result)

        _status = d.pop("status", UNSET)
        status: Union[Unset, DqoJobStatus]
        if isinstance(_status, Unset):
            status = UNSET
        else:
            status = DqoJobStatus(_status)

        collect_error_samples_result = cls(
            job_id=job_id,
            result=result,
            status=status,
        )

        collect_error_samples_result.additional_properties = d
        return collect_error_samples_result

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/execution/errorsampling/jobs/CollectErrorSamplesResult.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.execution.errorsampling.jobs;

import com.dqops.core.jobqueue.DqoQueueJobId;
import com.dqops.core.jobqueue.monitoring.DqoJobStatus;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import io.swagger.annotations.ApiModel;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * Object returned from the operation that queues a "collect error samples" job. The result contains the job id that was started
 * and optionally can also contain the result of collecting the error samples if the operation was started with wait=true parameter to wait for the "collect error samples" job to finish.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@ApiModel(value = "CollectErrorSamplesResult", description = "Object returned from the operation that queues a \"collect error samples\" job. " +
        "The result contains the job id that was started and optionally can also contain the result of collecting the error samples  " +
        "if the operation was started with wait=true parameter to wait for the \"collect error samples\" job to finish.")
@EqualsAndHashCode(callSuper = false)
@Data
public class CollectErrorSamplesResult {
    /**
     * Job id that identifies a job that was started on the DQOps job queue.
     */
    @JsonPropertyDescription("Job id that identifies a job that was started on the DQOps job queue.")
    private DqoQueueJobId jobId;

    /**
     * Optional result object that is returned only when the wait parameter was true and the "collect error samples" job has finished.
     * Contains the summary result of the error samplers executed, including the number of samples collected.
     * The calling code (the data pipeline) can decide if further processing should be continued.
     */
    @JsonPropertyDescription("Optional result object that is returned only when the wait parameter was true and the \"collect error samples\" job has finished. " +
            "Contains the summary result of collecting error samples, including the number of error samplers that were executed, and the number of error samples collected. ")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private ErrorSamplerResult result;

    /**
     * Job status.
     */
    @JsonPropertyDescription("Job status")
    private DqoJobStatus status = DqoJobStatus.queued;


    public CollectErrorSamplesResult() {
    }

    /**
     * Creates a new model given all parameters.
     * @param jobId Job id.
     * @param result Job result.
     * @param status Job status.
     */
    public CollectErrorSamplesResult(DqoQueueJobId jobId, ErrorSamplerResult result, DqoJobStatus status) {
        this.jobId = jobId;
        this.result = result;
        this.status = status;
    }

    /**
     * Creates a new model given all parameters.
     * @param jobId Job id.
     */
    public CollectErrorSamplesResult(DqoQueueJobId jobId) {
        this.jobId = jobId;
    }

    public static class CollectErrorSamplesQueueJobResultSampleFactory implements SampleValueFactory<CollectErrorSamplesResult> {
        @Override
        public CollectErrorSamplesResult createSample() {
            return new CollectErrorSamplesResult() {{
                setJobId(new DqoQueueJobId.DqoQueueJobIdSampleFactory().createSample());
                setStatus(DqoJobStatus.finished);
                setResult(new ErrorSamplerResult() {{
                    setExecutedErrorSamplers(3);
                    setColumnsAnalyzed(1);
                    setColumnsSuccessfullyAnalyzed(0);
                    setTotalErrorSamplersFailed(1);
                    setTotalErrorSamplesCollected(2);
                }});
            }};
        }
    }
}
